docs/data/data-grid/column-dimensions/ColumnAutosizing.tsx

```diff
@@ -0,0 +1,130 @@
+import * as React from 'react';
+import {
+  Button,
+  Checkbox,
+  FormControlLabel,
+  Rating,
+  Stack,
+  TextField,
+} from '@mui/material';
+import { useGridApiRef } from '@mui/x-data-grid';
+import {
+  DataGridPro,
+  GridApiPro,
+  DEFAULT_GRID_AUTOSIZE_OPTIONS,
+} from '@mui/x-data-grid-pro';
+import { randomRating, randomTraderName } from '@mui/x-data-grid-generator';
+
+function renderRating(params: any) {
+  return <Rating readOnly value={params.value} />;
+}
+
+function useData(length: number) {
+  return React.useMemo(() => {
+    const names = [
+      'Nike',
+      'Adidas',
+      'Puma',
+      'Reebok',
+      'Fila',
+      'Lululemon Athletica Clothing',
+      'Varley',
+    ];
+    const rows = Array.from({ length }).map((_, id) => ({
+      id,
+      brand: names[id % names.length],
+      rep: randomTraderName(),
+      rating: randomRating(),
+    }));
+
+    const columns = [
+      { field: 'id', headerName: 'Brand ID' },
+      { field: 'brand', headerName: 'Brand name' },
+      { field: 'rep', headerName: 'Representative' },
+      { field: 'rating', headerName: 'Rating', renderCell: renderRating },
+    ];
+
+    return { rows, columns };
+  }, [length]);
+}
+
+export default function ColumnAutosizing() {
+  const apiRef = useGridApiRef<GridApiPro>();
+  const data = useData(100);
+
+  const [includeHeaders, setIncludeHeaders] = React.useState(
+    DEFAULT_GRID_AUTOSIZE_OPTIONS.includeHeaders,
+  );
+  const [excludeOutliers, setExcludeOutliers] = React.useState(
+    DEFAULT_GRID_AUTOSIZE_OPTIONS.excludeOutliers,
+  );
+  const [outliersFactor, setOutliersFactor] = React.useState(
+    String(DEFAULT_GRID_AUTOSIZE_OPTIONS.outliersFactor),
+  );
+  const [expand, setExpand] = React.useState(DEFAULT_GRID_AUTOSIZE_OPTIONS.expand);
+
+  return (
+    <div style={{ width: '100%' }}>
+      <Stack
+        spacing={2}
+        direction="row"
+        alignItems="center"
+        sx={{ marginBottom: '1em' }}
+      >
+        <div>
+          <Button
+            variant="outlined"
+            onClick={() =>
+              apiRef.current.autosizeColumns({
+                includeHeaders,
+                excludeOutliers,
+                outliersFactor: Number.isNaN(parseFloat(outliersFactor))
+                  ? 1
+                  : parseFloat(outliersFactor),
+                expand,
+              })
```

Comment:
Would be great to pass the same options through `autosizeOptions` prop, so that same options apply when double-clicking the resize handle 👍🏻

Reply:
About `autosizeOptions.columns` property, I think the behaviour might be unexpected in some use-cases, I tried to describe it a bit in https://github.com/mui/mui-x/pull/10180#discussion_r1328137399
What do you think?